drivers/disk/usdhc.c

```diff
@@ -2634,35 +2634,42 @@ static int usdhc_board_access_init(struct usdhc_priv *priv)
 	}
 
 	if (!priv->detect_gpio) {
-		LOG_INF("USDHC detection other than GPIO not implemented!");
-		return 0;
-	}
+		LOG_INF("USDHC detection other than GPIO");
+		/* DATA3 does not monitor card insertion */
+		base->PROT_CTRL &= ~USDHC_PROT_CTRL_D3CD_MASK;
+		if ((base->PRES_STATE & USDHC_PRES_STATE_CINST_MASK) != 0) {
+			priv->inserted = true;
+			ret = 0;
```

Comment:
just curious but why is this needed if you are going to change the declare of ret to be initialized to 0?

Reply:
Thanks for catching this, I have removed this line.